python/lsst/ap/association/assoc_db_sqlite.py

```diff
@@ -0,0 +1,479 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST/AURA.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+import sqlite3
+
+from lsst.meas.algorithms.indexerRegistry import IndexerRegistry
+import lsst.afw.table as afwTable
+import lsst.afw.geom as afwGeom
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from .dia_collection import *
+from .dia_object import *
+
+
+__all__ = ["AssociationDBSqliteConfig", "AssociationDBSqliteTask"]
+
+
+afw_to_db_types = {
+    "L": "INTEGER",
+    "Angle": "REAL"
+}
+
+
+class AssociationDBSqliteConfig(pexConfig.Config):
+    """!
+    \anchor AssociationDBSqliteConfig_
+
+    \brief Configuration parameters for the AssociationDBSqliteTask
+    """
+    db_name = pexConfig.Field(
+        dtype=str,
+        doc='Location on disk and name of the sqlite3 database for storing '
+        'and loading DIASources and DIAObjects.',
+        default=':memory:'
+    )
+    indexer = IndexerRegistry.makeField(
+        doc='Select the spatial indexer to use within the database.',
+        default='HTM'
+    )
```

Comment:
I'm copying a bit of the configuration from Ingest/LoadIndexedReferenceTask which does allow the indexer to be a configurable. I think there is no need to hard code a specific indexer here especially since the standard in the rest of the stack seems to make this a configurable. One thing that I may want this it is a configurable, though, is force a check on the task __init__ that the indexer specified in the config is the same as the one used on creation of the database (and also remove all the references to HTM). I'll work on this for the next commit.

Reply:
I don't think I saw any use of spatial indexing when you defined the DB, but that does sound like a reasonable approach if you can make the queries indexer-agnostic.